lib/internal/test_runner/mock.js

```diff
@@ -157,18 +161,40 @@ class MockTracker {
         'options.setter', setter, "cannot be used with 'options.getter'"
       );
     }
+    function getOriginalObject(descriptor) {
+      let original;
+
+      if (getter) {
+        original = descriptor?.get;
+      } else if (setter) {
+        original = descriptor?.set;
+      } else {
+        original = descriptor?.value;
+      }
+
+      return original;
+    }
 
-    const descriptor = ObjectGetOwnPropertyDescriptor(object, methodName);
-    let original;
-
-    if (getter) {
-      original = descriptor?.get;
-    } else if (setter) {
-      original = descriptor?.set;
-    } else {
-      original = descriptor?.value;
+    function findMethodOnPrototypeChain(instance, method, isDescriptor = false) {
+      const original = isDescriptor ?
+        instance :
+        ObjectGetOwnPropertyDescriptor(instance, method);
+
+      if (original && !isDescriptor) {
+        return original;
+      }
+
+      const proto = ObjectGetPrototypeOf(instance);
+      const desc = ObjectGetOwnPropertyDescriptor(proto, method);
+      if (!desc && proto.name) {
```

Comment:
Why `proto.name` would need to be truthy? I think this deserves a comment.

Reply:
Sure! I added it with an additional check but it's not required to make it work so I removed it